langchain/chains/combine_documents/map_reduce.py

```diff
@@ -49,14 +55,79 @@ def get_default_document_variable_name(cls, values: Dict) -> Dict:
                 )
         return values
 
-    def combine_docs(self, docs: List[Document], **kwargs: Any) -> str:
-        """Combine by mapping first chain over all, then stuffing into final chain."""
+    def compute_tokens(self, results: List, docs: List[Document], **kwargs: Any) -> int:
+        """Checks to see if, after combining a set of documents into one text, if it is too long."""
+        question_result_key = self.llm_chain.output_key
+        result_docs = [
+            Document(page_content=r[question_result_key], metadata=docs[i].metadata)
+            # This uses metadata from the docs, and the textual results from `results`
+            for i, r in enumerate(results)
+        ]
+        doc_dicts = []
+        for doc in result_docs:
+            base_info = {"page_content": doc.page_content}
+            base_info.update(doc.metadata)
+            document_info = {
+                k: base_info[k] for k in self.document_prompt.input_variables
+                # TODO: how does this work with plain QA, as opposed to this which is designed to work with
+                # QA with sources right now?
+            }
+            doc_dicts.append(document_info)
+        # Format each document according to the prompt
+        doc_strings = [self.document_prompt.format(**doc) for doc in doc_dicts]
+        combined_texts = "\n\n".join(doc_strings)
+
+        # tokenize the text and compute num tokens
+        tokenizer = GPT2TokenizerFast.from_pretrained('gpt2')
```

Comment:
since this is already on the llm class, maybe we can use self.combine_doc_chain.llm.get_num_tokens or the like

Reply:
This is done in my next commit. It's:

`self.combine_document_chain.llm_chain.llm.get_num_tokens()`

Reply:
Here is the commit https://github.com/hwchase17/langchain/commit/dd6d89322337d74a40843794362618a54b9a3cf5